src/masternodes/mn_checks.cpp

```diff
@@ -2645,6 +2645,14 @@ class CCustomTxApplyVisitor : public CCustomTxVisitor
         for (const auto& kv : obj.amounts.balances)
         {
             DCT_ID tokenId = kv.first;
+            auto paybackAmount = kv.second;
+
+            if (height >= Params().GetConsensus().FortCanningHillHeight && kv.first == DCT_ID{0})
+            {
+                tokenId = mnview.GetToken("DUSD")->first;
+                paybackAmount = mnview.GetAmountInCurrency(paybackAmount, {"DFI","USD"});
```

Comment:
The DFIP contained a 1% penalty fee for paying back a DUSD loan in DFI. I cannot see this in the code. Shouldn't this be accounted for here?

Reply:
Thanks, the penalty fee will be added from gov variable. It is still a WIP.